src/middlewared/middlewared/plugins/truecommand/wireguard.py

```diff
@@ -84,6 +84,15 @@ async def wireguard_connection_health(self):
                     # We never established handshake with TC if timestamp is 0, otherwise it's been more
                     # then 30 minutes, error out please
                     health_error = True
+                else:
+                    # It's possible that IP of TC changed and we just need to get up to speed with the
+                    # new IP. So if we have a correct handshake, we should ping the TC IP to see if it's
+                    # still reachable
+                    config = await self.middleware.call('datastore.config', 'system.truecommand')
+                    cp = await run(['ping', '-c', '3', '-t', '5', '-q', str(config['remote_address'])], check=False)
```

Comment:
why -c 3?

Reply:
if you remove `-c 3` and keep `-t 5` it means the TC can't be more than 1 hop away from us and that single packet has to be successful. Why not just do a `-c 4 or 5` ?

Reply:
I am not sure I follow ? Idea is to keep pinging for a maximum of 5 seconds. If we do `-c 4/5`, that can potentially take more time then 5 seconds if the IP is not reachable

Reply:
Hah, so -t on Debian is TTL setting and on freebsd is timeout setting......a subtle difference that changes the behavior. Disregard 😬

Reply:
good catch ; ) I updated the PR to make sure it works on debian as well. Thank you for pointing it out